app/models/miq_ae_instance.rb

```diff
@@ -136,6 +136,12 @@ def self.get_homonymic_across_domains(user, fqname, enabled = nil)
     MiqAeDatastore.get_homonymic_across_domains(user, ::MiqAeInstance, fqname, enabled)
   end
 
+  def self.get_homonymic_across_domains_noprefix(user, path, enabled = nil)
```

Comment:
@mkanoor Do you think this should be exposed as a separate method from `get_homonymic_across_domains` to the caller or does it make sense to have a single method for the caller and have it take a parameter?

I was thinking we retain the original method name as the exposed method to callers and use a keyword argument for `prefix`, where the default is true.  (`get_homonymic_across_domains_noprefix` would become a private method.)

Something like:
```ruby
def self.get_homonymic_across_domains(user, fqname, enabled = nil, prefix: true)
  return get_homonymic_across_domains_noprefix(user, path, enabled) unless prefix

  MiqAeDatastore.get_homonymic_across_domains(user, ::MiqAeInstance, fqname, enabled)
end

def self.get_homonymic_across_domains_noprefix(user, path, enabled = nil)
...
```

Since `prefix` is a keyword argument the caller **is not required** to pass the `enabled` parameter if they want set `prefix`.  Meaning, someone could call:
```ruby
get_homonymic_across_domains(user, fqname, :prefix => true)
```
and the `enabled` parameter would still default to `nil` as expected.

Reply:
@gmcculloug Your suggestion would lead to a better implementation